demo/addons/godot-openxr/editor/OpenXRRunSelect.gd

```diff
@@ -0,0 +1,86 @@
+tool
+extends OptionButton
+
+var available_runtimes : Array = Array()
+var runtime_select = null
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	var platform = OS.get_name()
+	runtime_select = preload("res://addons/godot-openxr/OpenXRRuntimeSelect.gdns")
+	if runtime_select:
+		runtime_select = runtime_select.new()
+
+	var current_runtime = -1
+	if platform == "Windows":
+		var f = File.new()
+		if (f.open("res://addons/godot-openxr/runtimes.json", File.READ)) == OK:
+			var json = JSON.parse(f.get_as_text())
+			available_runtimes = json.result as Array
+			f.close()
+
+		var current_path = runtime_select.get_runtime()
+
+		for i in available_runtimes.size():
+			if available_runtimes[i]["path"] == current_path:
+				current_runtime = i
+
+		if (current_path != "" and current_runtime == -1):
+			# couldn't find it? add it to our list
+			var runtime_name = "???"
+
+			var pos = current_path.find_last("\\");
+			if pos != -1:
+				runtime_name = current_path.substr(pos+1)
+
+			available_runtimes.push_back({ "name": runtime_name, "path": current_path})
+			current_runtime = available_runtimes.size() - 1
+
+			# Maybe save our JSON?!?
+
+		# print("%s %s %d" % [available_runtimes, current_path, current_runtime])
+
+	if available_runtimes.size() > 0:
+		var dir = Directory.new()
+
+		clear()
+		for i in available_runtimes.size():
+			var runtime = available_runtimes[i]
+			if dir.file_exists(runtime["path"]):
+				add_item(runtime["name"], i)
+
+		selected = current_runtime
+
+		visible = true
+	else:
+		# I guess nothing supported on this platform
+		visible = false
+
+func _on_OpenXRRunSelect_item_selected(index):
+	if runtime_select:
+		print("Switching to " + available_runtimes[index]["path"])
+
+		# this currently doesn't work due to rights issues
+		# runtime_select.set_runtime(available_runtimes[index]["path"])
+
+		# this does but only in the editor and it's a little hacky
+		var runtime = available_runtimes[index]["path"]
+		var command = 'Set-ItemProperty -Path \'HKLM:\\SOFTWARE\\Khronos\\OpenXR\\1\' -Name \'ActiveRuntime\' -Value \'' + runtime + '\''
+		# var command = "Write-Host 'Test'"
+
+		# powershell wants unicode16, for now this is a DIRTY work around
+		# utf16 support is coming
```

Comment:
This is the only part I'm not happy about but for now I don't think we can do anything but accept it, unless someone wants to write a UTF-8 to UTF16LE parser in GDScript.
I'm keeping an eye out for Bruvzg work and whether a utf16 option becomes available in GDScript or I might add something myself in the not to distant future. Either way I suggest we handle this in a future PR.